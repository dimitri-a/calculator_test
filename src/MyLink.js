import React, {Component} from 'react';

const CLASS = {
    NORMAL: 'normal'
};

export default class MyLink extends Component {

    constructor(props) {
        super(props);

        this.state = {class: CLASS.NORMAL}

        this.onMouseEnter = this.onMouseEnter.bind(this);
    }

    onMouseEnter() {
        console.log('on hover');
        this.setState({class: CLASS.HOVER});

    }

    render() {

        return (<a href=""
                className = {this.state.class}
                onMouseEnter = {this.onMouseEnter}>{this.props.children}
            </a>
        );
    }

}
