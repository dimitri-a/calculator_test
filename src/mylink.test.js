
import React, { Component } from 'react';

import MyLink  from './MyLink.js';

import renderer from 'react-test-renderer';

test('test mylink', () => {

    const component = renderer.create(
        <MyLink page="http://www.facebook.com">Facebook</MyLink>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onMouseEnter()
    expect(tree).toMatchSnapshot();

});