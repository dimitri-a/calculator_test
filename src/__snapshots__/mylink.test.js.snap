// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mylink 1`] = `
<a
  className="normal"
  href=""
  onMouseEnter={[Function]}
>
  hier 
  Facebook
</a>
`;

exports[`test mylink 2`] = `
<a
  className="normal"
  href=""
  onMouseEnter={[Function]}
>
  hier 
  Facebook
</a>
`;
